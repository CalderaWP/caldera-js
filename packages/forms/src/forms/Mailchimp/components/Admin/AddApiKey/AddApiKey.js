import React, {Fragment,useState} from 'react'
import {Field} from '@calderajs/components';
const field = {
	fieldType: 'text',
	value: '',
	label: 'New API Key',
	fieldId: 'caldera-mc-api-key',
	required: true
};
/**
 * Add Mailchimp API Key
 *
 * @param listFieldConfig
 * @param listId
 * @param setList
 * @param instanceId
 * @return {*}
 * @constructor
 */
export const AddApiKey = ({onChange,instanceId,apiKey,onSaveApiKey}) => {
	return(
		<Fragment>
			<Field
				field={field}
				onChange={(newValue)=>{
					onChange(newValue);
				}}
				value={apiKey}
				instanceId={`caldera-mc-select-${instanceId}` }
			/>
			<button
				onClick={onSaveApiKey}
				id={`${instanceId}-mc-add-api-key`}

				title={'Save API Key'}
			>
				Save API Key
			</button>

		</Fragment>

	)

}

