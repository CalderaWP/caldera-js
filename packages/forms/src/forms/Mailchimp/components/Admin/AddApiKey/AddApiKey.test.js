import React from 'react';
import {mount} from 'enzyme/build';
import { render, fireEvent, getByTestId} from "react-testing-library";
import {AddApiKey} from "./AddApiKey";

const field = [{"fieldType":"select","required":true,"fieldId":"mc-select-field","options":[{"value":"45907f0c59","label":"Future Capable"}]}];
it( 'Shows list', () => {
	expect( render(
		<AddApiKey
			listFieldConfig={field}
			listId={'45907f0c59'}
			setList={()=>{}}
			instanceId={'a'}
		/>
	) ).toMatchSnapshot();
});

it( 'Changes list', () => {
	const onChange = jest.fn();
	const component =  mount(
		<AddApiKey
			listFieldConfig={field[0]}
			listId={''}
			onChange={onChange}
			instanceId={'test'}
		/>
	);

	component.find( '#caldera-mc-select-test' ).first().simulate(
		'change',
		{target: {value:'45907f0c59'}}
	);
	expect( onChange.mock.calls.length).toBe(1)

});
