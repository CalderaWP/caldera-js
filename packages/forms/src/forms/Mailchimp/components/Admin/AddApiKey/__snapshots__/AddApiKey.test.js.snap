// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows list 1`] = `
Array [
  <div
    className="caldera-field-wrapper caldera-field-wrapper-text"
  >
    <label
      className="caldera-field-label caldera-field-label-text"
      htmlFor="caldera-mc-api-key"
    >
      New API Key
       
    </label>
    <input
      className="caldera-field caldera-field-text"
      id="caldera-mc-api-key"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <button
    id="a-mc-add-api-key"
    title="Save API Key"
  >
    Save API Key
  </button>,
]
`;
